### PR TITLE
acpype

### PR DESCRIPTION
**Pull request by simonbray**
Creating GROMACS topologies for small molecules using Ambertools/acpype.

### CODE REVIEW

File: tools/buildtools/ambertools/acpype.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            <option value="gaff">gaff</option>
+            <option value="amber">AMBER14SB</option>
+            <option value="gaff2">gaff2</option>
+            <option value="amber2">AMBER14SB + gaff2</option>
+        </param>
+        <param name="save_gro" type="boolean" value="false" label="Save GRO file?" help="Save atomic coordinates of ligand in GRO format."/>
+    </inputs>
+    <outputs>
+        <data format="itp" name="output" label="Topology" from_work_dir='base.acpype/base_GMX.itp'/>
+        <data name="gro_output" format="gro" from_work_dir="base.acpype/base_GMX.gro">
```
Comment (chrisbarnettster) [inline tools/buildtools/ambertools/acpype.xml]: Also label this output.

File: tools/buildtools/ambertools/acpype.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<tool id="acpype" name="Generate MD topologies for small molecules" version="0.1.0">
```
Comment (chrisbarnettster) [inline tools/buildtools/ambertools/acpype.xml]: track conda package version rather? I think we have been generally tracking against the conda package version and then would add minor codes if the tool xml changes a great deal.

File: tools/buildtools/ambertools/acpype.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    <has_line line="[ atoms ]"/>
+                    <has_text text="    15   ha     1   JZ4    H4   15     0.13"/>
+                    <has_text text="    18   hc     1   JZ4    H7   18     0.05"/>
+                </assert_contents>
+            </output>
+            <output name="gro_output" file="base_GMX.gro" lines_diff="2"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+Tool to produce GROMACS topologies for small molecules using AmberTools.
```
Comment (chrisbarnettster) [inline tools/buildtools/ambertools/acpype.xml]: using Ambertools via the acpype interface,

File: tools/buildtools/ambertools/acpype.xml (hunk 2 above)
Comment (simonbray) [reply tools/buildtools/ambertools/acpype.xml]: I'm not a huge fan of `2019.03.22.14.36` as a version name, but you're right, it's better to be consistent.

**Comment by simonbray** [comment]
Thanks for the comments, I've updated with a new commit.

On another point, it's apparently also possible to use acpype to generate CHARMM topologies. I haven't really used CHARMM, so I don't know if this is interesting for us, but perhaps you have a view?

**Comment by chrisbarnettster** [comment]
> Thanks for the comments, I've updated with a new commit.
> 
> On another point, it's apparently also possible to use acpype to generate CHARMM topologies. I haven't really used CHARMM, so I don't know if this is interesting for us, but perhaps you have a view?

Yes this is interesting. Maybe to be added as an issue for future dev. I think Tharindu uses alternative options. @tsenapathi what do you think?

File: tools/buildtools/ambertools/.shed.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+name: ambertools
+owner: chemteam
+description: "Ambertools is a set of packages for preparing systems for molecular dynamics (MD) simulations and analyzing trajectories."
+homepage_url: http://ambermd.org/AmberTools.php
+long_description: |
+    AmberTools (http://ambermd.org/AmberTools.php) is a set of packages than can be to create forcefields (antechamber), prepare molecules(tleap) and much more.
+remote_repository_url: https://github.com/galaxycomputationalchemistry/galaxy-tools-compchem/
+type: unrestricted
+categories:
+- Molecular Dynamics
```
Comment (bgruening) [inline tools/buildtools/ambertools/.shed.yml]: indentation ...

File: tools/buildtools/ambertools/acpype.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+<tool id="acpype" name="Generate MD topologies for small molecules" version="2019.03.22.14.36">
```
Comment (bgruening) [inline tools/buildtools/ambertools/acpype.xml]: using a token here ...

File: tools/buildtools/ambertools/acpype.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        acpype -i './input1.${input1.ext}'
+            -n '$charge'
+            -b base
+            -m '$multiplicity'
+            -a '$atomtype'
+            -o gmx
+    ]]></command>
+    <inputs>
+        <param name="input1" type="data" format="pdb,mol2" label="Input file"
+                        help="Structure of the compound in pdb or mol2 format"/>
```
Comment (bgruening) [inline tools/buildtools/ambertools/acpype.xml]: indentation looks off here

File: tools/buildtools/ambertools/acpype.xml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+.. class:: infomark
+
+**Outputs**
+
+GROMACS topology for the ligand, in itp format. Optional: structure file, in gro format.
+
+
+    ]]></help>
+    <citations>
```
Comment (bgruening) [inline tools/buildtools/ambertools/acpype.xml]: you can use the marco from the other PR, isn't it?

Comment (simonbray) [reply tools/buildtools/ambertools/acpype.xml]: Actually I think it's not worth using the macro. This one needs acpype as well as ambertools so requirements and citations are different.

**Comment by bgruening** [comment]
This now needs an update.

File: tools/buildtools/ambertools/acpype.xml (hunk 7 above)
Comment (bgruening) [reply tools/buildtools/ambertools/acpype.xml]: And the version?

**Comment by bgruening** [comment]
Merci!

**Pull request merged**